### PR TITLE
[feat] added cloudflare static tunnels support

### PR DESCRIPTION
**Pull request by waruhachi**
This pertains to https://github.com/Wave-Play/robo.js/issues/276

### CODE REVIEW

File: packages/robo/src/cli/utils/cloudflared.ts [hunk 1]
```diff
@@ -200,3 +329,87 @@ function download(url: string, to: string, redirect = 0): Promise<string> {
...
+	return data;
+}
+
+async function updateTunnelConfig(id: string, accountId: string) {
+	const tunnelConfig: TunnelConfig = {
+		"config": {
+			"ingress": [
+				{
+					"hostname": `robo.${process.env.CLOUDFLARE_DOMAIN}`,
+					"service": "http://localhost:3000"
```
Comment (Pkmmte) [inline packages/robo/src/cli/utils/cloudflared.ts:360]: Local service port should be `process.env.PORT` by default if that exists.

File: packages/robo/src/cli/utils/cloudflared.ts [hunk 2]
```diff
@@ -200,3 +329,87 @@ function download(url: string, to: string, redirect = 0): Promise<string> {
...
+		envContent = envContent.replace(regex, `${key}="${value}"`);
+	} else {
+		envContent += `\n${key}="${value}"`;
+	}
+
+	fs.writeFileSync(path.join(process.cwd(), '.env'), envContent, 'utf8');
+	cloudflareLogger.debug(`Updated .env file with ${key}=${value}`)
+}
+
+function installService(token: string) {
```
Comment (Pkmmte) [inline packages/robo/src/cli/utils/cloudflared.ts:392]: Do we really want to install `cloudflared` as a service on the user machine even though `startCloudflared` already runs the tunnel associated with that token?

File: packages/robo/src/cli/utils/cloudflared.ts [hunk 3]
```diff
@@ -73,22 +152,72 @@ export function isCloudflaredInstalled(to = DEFAULT_BIN_PATH): boolean {
...
+		return true;
+	} catch (error) {
+		cloudflareLogger.error('Failed to initialize Cloudflare tunnel:', error);
+		return false;
+	}
+}
+
 export function startCloudflared(url: string) {
+	const tunnelId = process.env.CLOUDFLARE_TUNNEL_ID;
+	const tunnelToken = process.env.CLOUDFLARE_TUNNEL_TOKEN;
```
Comment (Pkmmte) [inline packages/robo/src/cli/utils/cloudflared.ts:198]: Can we reduce the number of items requested from the user by requesting the tunnel token from Cloudflare ourselves?

File: packages/robo/src/cli/utils/cloudflared.ts [hunk 4]
```diff
@@ -73,22 +152,72 @@ export function isCloudflaredInstalled(to = DEFAULT_BIN_PATH): boolean {
...
+	const tunnelDomain = process.env.CLOUDFLARE_DOMAIN;
+
+	const commandArgs = (tunnelId && tunnelToken && tunnelDomain) ? ['tunnel', '--no-autoupdate', 'run', '--token', tunnelToken, '--url', url, tunnelId] : ['tunnel', '--url', url, '--no-autoupdate'];
```
Comment (Pkmmte) [inline packages/robo/src/cli/utils/cloudflared.ts:201]: This ternary operation is a bit tough to read quickly. Mind cleaning it by appending the token conditionally instead? The code overall would be longer but more legible.

File: packages/robo/src/cli/utils/cloudflared.ts [hunk 5]
```diff
@@ -73,22 +152,72 @@ export function isCloudflaredInstalled(to = DEFAULT_BIN_PATH): boolean {
...
 
+export async function initializeCloudflareTunnel(): Promise<boolean> {
+	if (!process.env.CLOUDFLARE_API_KEY || !process.env.CLOUDFLARE_ACCOUNT_ID || !process.env.CLOUDFLARE_DOMAIN) {
+		cloudflareLogger.warn('Missing required Cloudflare environment variables.');
+		return false;
+	}
+
+	try {
+		if (process.env.CLOUDFLARE_TUNNEL_ID) {
+			const roboTunnel = await cloudflareRequest(`/accounts/${process.env.CLOUDFLARE_ACCOUNT_ID}/cfd_tunnel/${process.env.CLOUDFLARE_TUNNEL_ID}/token`);
```
Comment (Pkmmte) [inline packages/robo/src/cli/utils/cloudflared.ts:163]: Do Cloudflare's tunnel tokens expire at some point? If so, do we have a way to auto refresh them?

**Review by Pkmmte — CHANGES_REQUESTED** [review]
Looks pretty good! Just have a few questions on the execution of things.

Only required changes are using the `PORT` environment variable when applicable and running the new code through Prettier for format consistency.

File: packages/robo/src/cli/utils/cloudflared.ts (hunk 1 above)
Comment (waruhachi) [reply packages/robo/src/cli/utils/cloudflared.ts:360]: Fixed. Port will default to 3000 if none is set in the .env file

File: packages/robo/src/cli/utils/cloudflared.ts (hunk 3 above)
Comment (waruhachi) [reply packages/robo/src/cli/utils/cloudflared.ts:198]: Currently only the Account ID, API Key, and the domain are requested from the user. The tunnel ID and token are stored in the .env file after the initial tunnel creation for later use.

File: packages/robo/src/cli/utils/cloudflared.ts (hunk 4 above)
Comment (waruhachi) [reply packages/robo/src/cli/utils/cloudflared.ts:201]: Refactored for easier readability

**Pull request merged**